src/pages/home/ReportScreen.js

```diff
@@ -160,12 +156,29 @@ function ReportScreen({
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const {isSmallScreenWidth} = useWindowDimensions();
-
-    const firstRenderRef = useRef(true);
+    const {isOffline} = useNetwork();
     const flatListRef = useRef();
     const reactionListRef = useRef();
+    const firstRenderRef = useRef(true);
+    const reportIDFromRoute = getReportID(route);
+    const reportActionIDFromRoute = lodashGet(route, 'params.reportActionID', null);
+    const shouldTriggerLoadingRef = useRef(!!reportActionIDFromRoute);
     const prevReport = usePrevious(report);
     const prevUserLeavingStatus = usePrevious(userLeavingStatus);
+    const [isPrepareLinkingToMessage, setLinkingToMessage] = useState(!!reportActionIDFromRoute);
+
+    const reportActions = useMemo(() => {
+        if (!!allReportActions && allReportActions.length === 0) {
```

Comment:
Since you removed the Onyx selector, `allReportActions` is an object, not an array:

```suggestion
        if (_.isEmpty(allReportActions)) {
```

Reply:
Off-topic question: I remember we talked about getting rid of lodash. Is that still on the table, or are we just planning to get rid of the excessive methods?

Reply:
I got it. Thanks for the details!